### PR TITLE
Add slider to task updation form

### PR DESCRIPTION
**Pull request by swarajpure**
Picking up from #153

### CODE REVIEW

File: app/components/task/holder.hbs [hunk 1]
```diff
@@ -1,28 +1,9 @@
...
-    </select>
-  </div>
-  <div class="task-card__btn">
-    <button type="button" {{on "click" (fn @onTaskUpdate @task.id)}}>Update Task</button>
-  </div>
   <div class="progress-bar-container">
     <div class="progress-bar">
-      <div class="bar percent-{{@task.percentCompleted}}">
+      <input id="input" type="range" name="percentCompleted" value="{{@task.percentCompleted}}" min="0" max="100" step="25" {{on "change" this.onPercentageChange}} {{on "change" (fn @onTaskUpdate @task.id)}}
+      list="status"/>
```
Comment (harshith-venkatesh) [inline app/components/task/holder.hbs:6]: What is the significance of `list="status"` here?

File: app/styles/tasks.css [hunk 2]
```diff
@@ -36,49 +41,114 @@
...
-.percent-60 {
-  width: 60%;
+.input{
+  overflow: hidden;
 }
 
-.percent-80 {
-  width: 80%;
+input[type=range]:focus {
+  outline: none;
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:72]: I hope this won't be causing any issue for web accessibility and is tested however
Can we go for these options instead https://www.a11yproject.com/posts/2013-01-25-never-remove-css-outlines/

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: app/styles/tasks.css (hunk 2 above)
Comment (swarajpure) [reply app/styles/tasks.css]: Have removed it, thanks for pointing out

File: app/components/task/holder.hbs (hunk 1 above)
Comment (swarajpure) [reply app/components/task/holder.hbs]: Not required here, don't know why it was added

File: app/styles/tasks.css [hunk 3]
```diff
@@ -36,49 +41,107 @@
...
-    rgba(255, 255, 255, .2) 50%,
-    rgba(255, 255, 255, .2) 75%,
-    transparent 75%,
-    transparent
-  );
-  background-size: 30px 30px;
-  animation: move 2s linear infinite;
-  box-shadow: 2px 0 10px inset rgba(0,0,0,0.2);
-  transition: width 2s ease-out;
+input[type=range]::-moz-range-track {
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:92]: Eager to see what each css properties does here

File: app/styles/tasks.css [hunk 4]
```diff
@@ -1,3 +1,8 @@
+:root{
+  --slider-colour: #424CC1;
+  --slider-background: #CFD3FF;
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:3]: NIT: If we are repeating #A1D0FF could we think about keeping a element in root itself

**Review by harshith-venkatesh — DISMISSED** [review]

File: app/styles/tasks.css (hunk 3 above)
Comment (swarajpure) [reply app/styles/tasks.css:92]: https://css-tricks.com/styling-cross-browser-compatible-range-inputs-css/

File: app/styles/tasks.css (hunk 4 above)
Comment (swarajpure) [reply app/styles/tasks.css:3]: Fixed, thanks for pointing out :)

File: app/styles/tasks.css [hunk 5]
```diff
@@ -36,49 +42,116 @@
 
 /* progress bar */
 
+.wrapper-div{
+  margin: 1em;
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:46]: any reason why we are going with 2 different units here

File: app/styles/tasks.css [hunk 6]
```diff
@@ -36,49 +42,116 @@
...
 .progress-bar {
-  background-color: #BFADA3;
-  height: 15px;
-  border-radius: 20px;
+  background-color: #fff;
+  height: 40px;
+  border-radius: 25px;
   overflow: hidden;
-  box-shadow: 2px 0 10px inset rgba(0,0,0,0.2);
   margin-top: 16px;
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:55]: is it better to shift to rem?

File: app/styles/tasks.css [hunk 7]
```diff
@@ -36,49 +42,116 @@
...
 }
 
-.percent-60 {
-  width: 60%;
+input[type=range]::-webkit-slider-runnable-track {
+  width: 50%;
+  height: 5px;
+  cursor: pointer;
+  background-color: var(--slider-background);
+  box-shadow: 0px 0px 0px #000000;
```
Comment (harshith-venkatesh) [inline app/styles/tasks.css:73]: NIT: `#000` would be fine too right like how you mentioned `#fff` above

**Review by harshith-venkatesh — APPROVED** [review]
Rest all looks good, thanks for sharing the link

File: app/styles/tasks.css (hunk 6 above)
Comment (swarajpure) [reply app/styles/tasks.css:55]: rem makes sense for font-size, any specific reason to use it for margin?

**Pull request merged**